### PR TITLE
Minor kernel gpio/irq changes and a new GCC warning

### PR DESCRIPTION
**Pull request by OdysseusGE**
Two more minor changes that allow a clean compile with recent kernel and gcc versions, in addition to the one proposed by @dlunev.

### CODE REVIEW

**Comment by daniel-santos** [comment]
Thanks for putting this together. I've gotten behind a bit. Going to test these out now.

File: mcp2210-gpio.c [hunk 1]
```diff
@@ -93,7 +93,11 @@ int mcp2210_gpio_probe(struct mcp2210_device *dev)
 		return -EINVAL;
 
 	gpio->label		= "mcp2210";
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,5,0)
+	gpio->parent	= &dev->udev->dev;
```
Comment (daniel-santos) [inline mcp2210-gpio.c]: Please keep code formatting the same. We're using the same standard as Linux -- 8 spaces per tab, so `parent` needs another tab after it.

**Comment by daniel-santos** [comment]
I've merged a few of your commits with a little editing for formatting.  I don't recall if the Linux coding standard uses one-line if statements (I think it might), but I don't tend to so much.  I'll check the rest out once I setup my testing board again.  Thanks for your contributions!

**Comment by OdysseusGE** [comment]
Sorry, I'm a bit new to this and should have put those udev rule/script changes in their own branch to start with.  But you've added everything else I indented you to pull.  Thanks!

**Comment by daniel-santos** [comment]
Hey, no problem. I still feel new to this after a few years. :) Thanks again. I'll take a look at the udev scripts a bit later on, I seem to have come down with a cold or something, so not too much energy right now.

**Pull request closed without merging**